hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieFileWriteHandler.java

```diff
@@ -0,0 +1,115 @@
+ * limitations under the License.
+ */
+
+package org.apache.hudi.io;
+
+import org.apache.hadoop.fs.Path;
+import org.apache.hudi.client.WriteStatus;
+import org.apache.hudi.common.engine.TaskContextSupplier;
+import org.apache.hudi.common.fs.FSUtils;
+import org.apache.hudi.common.model.HoodieRecordPayload;
+import org.apache.hudi.common.model.IOType;
+import org.apache.hudi.config.HoodieWriteConfig;
+import org.apache.hudi.exception.HoodieInsertException;
+import org.apache.hudi.execution.ParquetFileMetaToWriteStatusConvertor;
+import org.apache.hudi.table.HoodieTable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Write handle that is used to work on top of files rather than on individual records.
+ */
+public class HoodieFileWriteHandler<T extends HoodieRecordPayload, I, K, O> extends HoodieWriteHandle<T, I, K, O> {
+
+  private static final Logger LOG = LoggerFactory.getLogger(HoodieFileWriteHandler.class);
+  private final Path path;
+  private String prevCommit;
+
+  public HoodieFileWriteHandler(HoodieWriteConfig config, String instantTime, HoodieTable<T, I, K, O> hoodieTable,
+                                String partitionPath, String fileId, TaskContextSupplier taskContextSupplier,
+                                Path srcPath) {
+    super(config, instantTime, partitionPath, fileId, hoodieTable, taskContextSupplier);
+
+    // Output file path.
+    this.path = makeNewPath(partitionPath);
+    this.prevCommit = srcPath.getName().split("_")[2].split("\\.")[0];
+
+    // Create inProgress marker file
+    createMarkerFile(partitionPath, FSUtils.makeBaseFileName(this.instantTime, this.writeToken, this.fileId, hoodieTable.getBaseFileExtension()));
+    // TODO: Create inprogress marker here and remove above marker file creation, once the marker PR is landed.
+    // createInProgressMarkerFile(partitionPath,FSUtils.makeDataFileName(this.instantTime, this.writeToken, this.fileId, hoodieTable.getBaseFileExtension()));
+    LOG.info("New CreateHandle for partition :" + partitionPath + " with fileId " + fileId);
+  }
+
+  /**
+   * Complete writing of the file by creating the success marker file.
+   * @return WriteStatuses, ideally it will be only one object.
+   */
+  @Override
+  public List<WriteStatus> close() {
+    LOG.info("Closing the file " + this.fileId + " as we are done with the file.");
+    try {
+      Map<String, Object> executionConfigs = new HashMap<>();
+      executionConfigs.put("prevCommit", prevCommit);
+      executionConfigs.put("timeTaken", timer.endTimer());
+
+      this.writeStatus = generateWriteStatus(path.toString(), partitionPath, executionConfigs);
+
+      // TODO: Create completed marker file here once the marker PR is landed.
+      // createCompleteMarkerFile throws hoodieException, if marker directory is not present.
```

Comment:
Not sure what these TODO really mean, the marker file should be created anyway.

Reply:
please do add jira id here

Reply:
This PR is dependent on Marker changes PR from Balajee. So, for now added these comments will revert as soon as those changes are landed.

Reply:
Added it as part of TODO statement.